### DISCUSSION TITLE
Some interoperability suggestions

### DISCUSSION THREAD

**Discussion by PegasusPlusUS** [nushell/nushell#14355] [Ideas]
Hi, I'm a new user of Nushell. I found some interoperability and user-friendly improvement potentials:

1. Consider supporting dir (GitHub Codespaces bash supports it, and PowerShell supports dir/ls).

2. When users try to use cat or type, they encounter error messages. Providing an alias for open might be better, or at least mentioning open in the current error message.

3. On Windows, currently, if using a drive letter, only absolute paths are allowed. However, Windows natively supports the "current directory" concept for each drive. I hope Nushell can also accept paths like C: or D:test.

**Comment by fdncred** [comment]
The first 2 can be handled with aliases like you mentioned. It's easy enough to place those in your config.nu file.

I'm not on a Windows system today but number 3 looks familiar. Can you explain more about what it's doing and what you want nushell to do?

**Reply by PegasusPlusUS** [reply]
Yeah 2 aliases is ok, I noticed on first nushell run, it generated default config file, wish it can have them as default, :).

3. Windows has 'current directory' on each driver, here's example in cmd.exe:
```cmd.exe snapshot
C:\D>d:

D:\>dir C:
 Volume in drive C has no label.
 Volume Serial Number is 7C5C-DB1A

 Directory of C:\D

11/15/2024  08:11 PM    <DIR>          .
11/02/2024  05:56 PM    <DIR>          NextestMinimalRepro
11/15/2024  06:04 PM    <DIR>          test
               0 File(s)              0 bytes
               3 Dir(s)  18,012,397,568 bytes free

D:\>dir c:..
 Volume in drive C has no label.
 Volume Serial Number is 7C5C-DB1A

 Directory of C:\

11/12/2024  02:30 AM    <DIR>          BitIntelligence
11/13/2024  08:16 PM    <DIR>          cygwin64
11/15/2024  08:11 PM    <DIR>          D
11/12/2024  07:24 PM    <DIR>          Intel
07/18/2023  01:32 PM    <DIR>          KuGou
05/06/2022  09:24 PM    <DIR>          PerfLogs
11/12/2024  02:41 AM    <DIR>          Program Files
11/08/2024  07:28 PM    <DIR>          Program Files (x86)
06/14/2022  04:25 PM    <DIR>          ReBase
04/29/2023  12:54 PM    <DIR>          Users
11/12/2024  07:23 PM    <DIR>          Windows
               0 File(s)              0 bytes
              11 Dir(s)  17,944,080,384 bytes free

D:\>
```

In nushell, only absolute directory can be used after driver letter:, and switch to another driver also need absolute path, can not use just driver letter:

```nushell snapshot
C:\> ls d:                                                                                                                                              11/15/2024 06:32:02 PMError:   x Main thread panicked.
  |-> at crates\nu-path\src\path.rs:1419:9
  `-> assertion failed: Form::invariants_satisfied(&buf)
  help: set the `RUST_BACKTRACE=1` environment variable to display a backtrace.

C:\> d:                                                                                                                                                 11/16/2024 10:08:12 AMError:   × Cannot set $env.PWD to a non-absolute path


C:\> D:/                                                                                                                                                11/16/2024 10:08:15 AM
```

**Comment by PegasusPlusUS** [comment]
![image](https://github.com/user-attachments/assets/4229b457-9e61-4507-9811-157bdce48c92)

**Reply by fdncred** [reply]
I could maybe see someone implementing the D: vs D:\. I'm not sure about the current directory thing because it would require a fundamental change where every drive would have a current working directory. But all it takes is someone passionate enough and talented enough to implement it.

**Reply by PegasusPlusUS** [reply]
Yeah, in windows, one can even change the current directory of another driver, but not switch there:

![Screenshot 2024-11-16 174310](https://github.com/user-attachments/assets/3f46d1ce-02e8-4281-813e-4b4e9d2ba514)

**Reply by NotTheDr01ds** [reply]
> Yeah, in windows, one can even change the current directory of another driver, but not switch there:

That behavior is in CMD only, but PowerShell actually switches there.  I've disliked the CMD approach for over 30 years now. ;-)

When a user says `cd C:\D`, they don't want to then have to type `c:` to get there if they weren't already.

**Comment by PegasusPlusUS** [comment]
Complete PWD per drive support for nushell, will send PR after testing several days.

![Screenshot 2024-11-19 050234](https://github.com/user-attachments/assets/6a12c164-5c09-46e3-a7ad-e4bf5894ad7b)

**Reply by PegasusPlusUS** [reply]
PR #14411
